clang-tools-extra/clang-tidy/modernize/UseEmplaceCheck.cpp

```diff
@@ -207,11 +211,13 @@ void UseEmplaceCheck::registerMatchers(MatchFinder *Finder) {
   auto HasConstructExpr = has(ignoringImplicit(SoughtConstructExpr));
 
   // allow for T{} to be replaced, even if no CTOR is declared
-  auto HasConstructInitListExpr = has(initListExpr(anyOf(
-      allOf(has(SoughtConstructExpr),
-            has(cxxConstructExpr(argumentCountIs(0)))),
-      has(cxxBindTemporaryExpr(has(SoughtConstructExpr),
-                               has(cxxConstructExpr(argumentCountIs(0))))))));
+  auto HasConstructInitListExpr =
+      has(initListExpr(anyOf(initCountIs(0), initCountIs(1)),
```

Comment:
`initCountIs(1)` will work only if we got constructor that take that argument.

```
#include <vector>

struct test { int a; };

int main()
{
    std::vector<test> aa;
    aa.emplace_back(1);

    return 0;
}
```

Code like this does not compile.

Reply:
This is needed for these existing use cases: https://github.com/llvm/llvm-project/pull/66169/files#diff-74fd330539ffeaec08a2fdb73b86cd2d31fcb90c2b2e934bc792901b3035534fR1296